static/js/global.js

```diff
@@ -0,0 +1,633 @@
+        $('#ta, #tr, #trm').each(function () {
+            v = util.CtoF($(this).val());
+            $(this).val(v.toFixed(1)).spinner({
+                step: envVarLimits.ta.ip.step,
+                min: envVarLimits.ta.ip.min,
+                max: envVarLimits.ta.ip.max,
+            });
+        });
+
+        // change the air velocity unit
+        $('#vel-unit, #vel-a-unit').html(' fpm');
+
+        // Convert the air velocity to fpm and change spinner limit
+        v = $('#vel').val();
+        $('#vel').val(v * 196.9).spinner({
+            step: envVarLimits.vel.ip.step,
+            min: envVarLimits.vel.ip.min,
+            max: envVarLimits.vel.ip.max,
+            numberFormat: 'n'
+        });
+
+        if (hs === 'dewpoint') {
+            $('#rh-unit').html(' &deg;F');
+            v = (util.CtoF($('#rh').val()));
+            $('#rh').val(v.toFixed(1)).spinner({
+                step: envVarLimits.tdp.ip.step,
+                min: envVarLimits.tdp.ip.min,
+                max: envVarLimits.tdp.ip.max,
+                numberFormat: "n"
+            });
+        } else if (hs === 'wetbulb') {
+            $('#rh-unit').html(' &deg;F');
+            v = (util.CtoF($('#rh').val()));
+            $('#rh').val(v.toFixed(1)).spinner({
+                step: envVarLimits.twb.ip.step,
+                min: envVarLimits.twb.ip.min,
+                max: envVarLimits.twb.ip.max,
+                numberFormat: "n"
+            });
+        } else if (hs === 'vappress') {
+            $('#rh-unit').html(' in HG');
+            v = $('#rh').val() / 2.953;
+            $('#rh').val(v.toFixed(2));
+        }
+    }
+    pc.toggleUnits(isCelsius);
+    bc.toggleUnits(isCelsius);
+    vc.toggleUnits(isCelsius);
+    ac.toggleUnits(isCelsius);
+}
+
+function updateGlobe() {
+    let ta = parseFloat($('#ta-g').val());
+    let vel = parseFloat($('#vel-g').val());
+    let tglobe = parseFloat($('#tglobe').val());
+    let diameter = parseFloat($('#diameter').val());
+    let emissivity = parseFloat($('#emissivity').val());
+    if (!isCelsius) {
+        ta = util.FtoC(ta);
+        vel /= 196.9;
+        tglobe = util.FtoC(tglobe);
+        diameter *= 0.0254
+    }
+    let tr = psy.globetemp(ta, vel, tglobe, diameter, emissivity);
+    if (!isCelsius) tr = util.CtoF(tr);
+    $('#mrt-result').val(tr.toFixed(1));
+}
+
+// check user entry
+function validateUserEntry(i) {
+
+    keys.forEach(function (element) {
+        d_cache[element] = d[element];
+        let e = document.getElementById(element + i).value;
+
+        // replace comma with dot
+        e = e.replace(/,/g, '.');
+
+        // define which measurement system is being used
+        let measurementSystem;
+        if (isCelsius) {
+            measurementSystem = 'si'
+        }
+        else {
+            measurementSystem ='ip'
+        }
+
+        // check user entry. if value is beyond the acceptability limit replace it with the default value
+        try {
```

Comment:
What is this try/catch for? It looks like it will fail silently if there's any problem in this block